layers/cmd/valueset.go

```diff
@@ -311,6 +319,109 @@ func LoadValuesetFiles(vs *Valuesets, files []string) error {
 	return nil
 }
 
+var re = regexp.MustCompile(`\([^()]*.?[^()]*\)$`)
+
+const (
+	sheetInputHeader  string = "input"
+	sheetOutputHeader string = "output"
+)
+
+func (vsets Valuesets) LoadSpreadsheets(ctx *ls.Context) error {
+	for _, spreadsheet := range vsets.Spreadsheets {
+		sheet, err := cmdutil.ReadSpreadsheetFile(spreadsheet)
+		if err != nil {
+			return fmt.Errorf("While reading file: %s, %w", spreadsheet, err)
+		}
+		sheetHeaders := make(map[string]string)
+		determineHeaderType := func(str string) (string, string) {
+			str = strings.TrimSpace(str)
+			e := re.FindString(str) // "(     i )"
+			headerType := strings.Trim(e, "(")
+			headerType = strings.Trim(e, ")")
+			switch strings.ToLower(strings.TrimSpace(headerType)) {
+			case "i":
+				return str[:strings.Index(str, "(")], sheetInputHeader
+			case "o":
+				return str[:strings.Index(str, "(")], sheetOutputHeader
+			}
+			return "", ""
+		}
+		for name, value := range sheet {
+			if _, exists := vsets.Sets[name]; exists {
+				return fmt.Errorf("Value set %s already defined -- in file: %s", name, sheet)
+			}
+			var headerIdx int
+			var headerFlag bool
+		ROWS:
+			for rowIdx, row := range value {
+				if len(row) == 0 {
+					continue
+				}
+				for colIdx, columnData := range row {
+					if !headerFlag {
+						_, headerType := determineHeaderType(columnData)
+						sheetHeaders[columnData] = headerType
+						headerIdx = rowIdx
+						if colIdx == len(row)-1 {
+							headerFlag = true
+						}
+						continue
+					}
+					columnHeader := value[headerIdx][colIdx]
+					if sheetHeaders[columnHeader] == sheetInputHeader || sheetHeaders[columnHeader] == sheetOutputHeader {
+						ioHeaderType(sheetHeaders, name, columnHeader, rowIdx, colIdx, value, &vsets)
+					} else {
+						kvPairHeaderType(name, columnHeader, rowIdx, colIdx, value, &vsets)
+						continue ROWS
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// spreadsheet - multiple k-v pairs as input. output the whole row
+func kvPairHeaderType(sheetName, columnHeader string, rowIdx, colIdx int, value [][]string, vsets *Valuesets) {
+	for _, opt := range vsets.Options.lookupInput {
```

Comment:
parsing a spreadsheet is not dependent on the options. Column headers are always keys, column value are always values.